src/main/java/io/airlift/units/ThreadsCount.java

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.units;
+
+import oshi.SystemInfo;
+
+import java.util.function.Supplier;
+
+import static io.airlift.units.Preconditions.checkArgument;
+import static java.lang.Math.min;
+import static java.lang.Math.round;
+import static java.lang.String.format;
+
+public class ThreadsCount
+        implements Comparable<ThreadsCount>
+{
+    private static final String PER_CORE_SUFFIX = "C";
+    private static final Supplier<Integer> AVAILABLE_PROCESSORS = MachineInfo::getAvailablePhysicalProcessorCount;
```

Comment:
Did you check https://github.com/trinodb/trino/pull/16303#discussion_r1122780546?

Reply:
Yes.

I've created a Dockerfile that runs a main method added to ThreadsCount:

```
 public static void main(String[] args)
 {
        System.out.println(ThreadsCount.valueOf("1.0C"));
 }
```

Dockerfile:
```
ARG JAVA_VERSION=17
FROM arm64v8/openjdk:${JAVA_VERSION}-ea-jdk

COPY target/units-1.9-SNAPSHOT.jar units.jar
COPY oshi-core.jar oshi-core.jar

CMD java -cp units.jar:oshi-core.jar io.airlift.units.ThreadsCount
```

Then:

```
docker build . -t threads-count
```

And results (8 cores are available to docker, this is ARM so no HT):

<img width="526" alt="Screenshot 2023-03-29 at 14 36 33" src="https://user-images.githubusercontent.com/66972/228537194-dbc81cb9-cb10-43b6-8026-d347d5d62093.png">

```
❯ docker run threads-count
8 threads
❯ docker run --cpus 2 threads-count
2 threads
❯ docker run --cpus 4 threads-count
4 threads
❯ docker run --cpus 2.5 threads-count
3 threads
```

Reply:
Does it answer your concerns @sopel39 ?

Reply:
I've copied that code from Trino, so physical cores are taken into the account for Intel

Reply:
cool!. thanks